### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by DaJansenGit**
* improves usage of pathlib (still some work to do)
* improves docs

### CODE REVIEW

File: CI_templates_python/README.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
 ## Create your own templates
 
-Execute the `python Dymola_python_tests/python CI_templates_python/ci_templates.py` command in the root directory of your repository. 
+Execute the `python python CI_templates_python/ci_templates.py` command in the root directory of your repository.
```
Comment (FWuellhorst) [inline CI_templates_python/README.md]: Duplicate python

File: CI_templates_python/README.md [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 The script will then ask you which tests and packages to check, adapting to your library. 
 
-Also the variables in the `Dymola_python_tests\CI_templates_python\ci_templates_configuration.py` should be checked before. 
-Important are the variables `image_name` and `variable_main_list`. These must be adapted to the current repo. The settings are then stored under `Dymola_python_tests\CI_templates_python\Setting\CI_setting.toml`. 
+Also the variables in the `CI_templates_python\ci_templates_configuration.py` should be checked before. 
+Important are the variables `image_name` and `variable_main_list`. These must be adapted to the current repo. The settings are then stored under `CI_templates_python\Setting\CI_setting.toml`. 
 If changes should be made in the settings, these can be made in the toml file. 
 
-Then the command `python Dymola_python_tests/python CI_templates_python/ci_templates.py --setting` must be executed. 
+Then the command `python python CI_templates_python/ci_templates.py --setting` must be executed.
```
Comment (FWuellhorst) [inline CI_templates_python/README.md]: Same here

**Review by FWuellhorst — CHANGES_REQUESTED** [review]
Minor first comments

File: CI_templates_python/README.md (hunk 1 above)
Comment (DaJansenGit) [reply CI_templates_python/README.md]: fixed

File: CI_templates_python/README.md (hunk 2 above)
Comment (DaJansenGit) [reply CI_templates_python/README.md]: fixed

File: CITests/CleanUp/structure_check.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 import sys
 from pathlib import Path
 import argparse
-
+# TODO Get rid of all references to "Dymola_python_tests"
 sys.path.append('Dymola_python_tests/CITests/CI_Configuration')
```
Comment (FWuellhorst) [inline CITests/CleanUp/structure_check.py]: Shouldn't this be fixed now?

File: README.md [hunk 4]
```diff
@@ -1,12 +1,68 @@
...
+![process](/docs/images/mocitempgen_process.png)
+* MoCITempGen uses the base CI-Templates in .txt form stored in `templates` 
+folder to create a library related CI-structure.
+* This can then be placed inside the target Modelica library
+* The CI tasks are executed by a GitLab runner which uses the functionality 
+provided through the python functions in `CITests` folder. This will be moved 
+to a public ModelicaPyCI package on PyPi soon
+* Through the
+[GitLab mirroring feature](https://docs.gitlab.com/ee/user/project/repository/mirror/) 
+not only GitLab repos can use MoCITempGen generated templates buy any GitHub,
```
Comment (FWuellhorst) [inline README.md]: buy -> but

File: CI_templates_python/ci_templates.py [hunk 5]
```diff
@@ -45,13 +47,14 @@ def _write_ci_structure_template(self):
                                       ci_interact_show_ref_file=self.ci_interact_show_ref_file,
                                       ci_interact_update_ref_file=self.ci_interact_update_ref_file,
                                       expire_in_time=self.expire_in_time)
-        ci_folder = f'{self.temp_dir}{os.sep}{self.temp_ci_structure_file.split(os.sep)[-2]}'
+        ci_folder = Path(self.temp_dir) / self.temp_ci_structure_file.parent.name
         CI_conf_class().check_ci_folder_structure([ci_folder])
-        yml_file = f'{ci_folder}{os.sep}{self.temp_ci_structure_file.split(os.sep)[-1]}'
-        yml_tmp = open(yml_file.replace(".txt", ".gitlab-ci.yml"), "w")
+        yml_file = (ci_folder / self.temp_ci_structure_file.stem).with_suffix('.yml')
```
Comment (FWuellhorst) [inline CI_templates_python/ci_templates.py:52]: / is not always save i think, use `.joinpath`  instead.

**Review by FWuellhorst — CHANGES_REQUESTED** [review]
Minot requests, looks good aside from that :)

**Pull request merged**